packages/core/src/external/aws/lambda-logic/document-bulk-signing.ts

```diff
@@ -0,0 +1,137 @@
+import { S3Utils } from "../s3";
+import { z } from "zod";
+import { DocumentReference } from "@medplum/fhirtypes";
+import { searchDocuments } from "../../opensearch/search-documents";
+import axios from "axios";
+import { capture } from "../../../util/notifications";
+import { getEnvVarOrFail } from "../../../util/env-var";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+const ossApi = axios.create();
+dayjs.extend(duration);
+
+const SIGNED_URL_DURATION_SECONDS = dayjs.duration({ minutes: 3 }).asSeconds();
+
+export type DocumentBulkSignerLambdaRequest = {
+  patientId: string;
+  cxId: string;
+  requestId: string;
+};
+
+export type DocumentBulkSignerLambdaResponse = {
+  id: string;
+  fileName: string;
+  description?: string;
+  mimeType?: string;
+  size?: number; // bytes
+  url: string;
+};
+
+export async function getSignedUrls(
+  cxId: string,
+  patientId: string,
+  requestId: string,
+  bucketName: string,
+  region: string,
+  apiURL: string
+) {
+  console.log(`DEBUGGING: fhir_server: ${getEnvVarOrFail("FHIR_SERVER_URL")}`);
+  //`search_endpoint: ${getEnvVarOrFail("SEARCH_ENDPOINT")}, search_index: ${getEnvVarOrFail("SEARCH_INDEX")}, search_username: ${getEnvVarOrFail("SEARCH_USERNAME")}, search_password: ${getEnvVarOrFail("PASSWORD")} `);
+  const s3Utils = new S3Utils(region);
+
+  const documents: DocumentReference[] = await searchDocuments({ cxId, patientId });
+
+  console.log(`Found ${documents}`);
+
+  const urls = await Promise.all(
```

Comment:
with `Promise.all`, this means if a single URL fails to sign, it will throw an exception:

- double checking this is the intention here?
- if so, we should probably wrap this in a try/catch
- on error, we should sent a sentry, and also alert the API of the failure, so we can update the status in the DB, and send a webhook accordingly

Reply:
good catch. I think the desired behavior is to not throw an exception if a single url fails. We should catch it and capture and then continue onto the next url. I added a try catch in `startBulkGetDocumentUrls` that actually calls the lambda, so if the lambda itself fails, then we will capture that too and update the query status to failed and return that to the cx.